src/Fmt.ml

```diff
@@ -150,15 +156,48 @@ let wrap_fits_breaks ?(space = true) conf x =
 
 (** Boxes ---------------------------------------------------------------*)
 
-let open_box n fs = Format.pp_open_box fs n
+let box_debug_enabled = ref false
+
+let box_stack = ref []
+
+let box_depth_colors = [|32; 33; 34; 31; 35; 36|]
+
+let box_depth_color () =
+  let depth = List.length !box_stack in
+  box_depth_colors.(depth % Array.length box_depth_colors)
+
+let debug_box_open open_sym close_sym fs =
+  if !box_debug_enabled then (
+    pp_color_k (box_depth_color ()) (fun fs -> fmt "@<0>%s" fs open_sym) fs ;
+    box_stack := close_sym :: !box_stack )
+
+let debug_box_close fs =
+  if !box_debug_enabled then
+    match !box_stack with
+    | [] -> pp_color_k 41 (fmt "@<0>]") fs
+    | close_sym :: tl ->
+        box_stack := tl ;
+        pp_color_k (box_depth_color ())
+          (fun fs -> fmt "@<0>%s" fs close_sym)
+          fs
+
+let open_box n fs =
+  debug_box_open "«" "»" fs ;
+  Format.pp_open_box fs n
 
-and open_vbox n fs = Format.pp_open_vbox fs n
+and open_vbox n fs =
+  debug_box_open "⟨" "⟩" fs ;
```

Comment:
Length of lines with unicode characters is wrong ?

Reply:
`@<0>` should take care of it if I understand your question correctly

Reply:
No. I mean there may be a bug in Ocamlformat:

```
let open_box n fs =
  debug_box_open "«" "»" fs ;
  Format.pp_open_box fs n

let open_box n fs = debug_box_open "<" ">" fs ; Format.pp_open_box fs n
```

Reply:
I see. Currently only comments are interpreted as if unicode as far as length goes. For actual code, I don't know what the right thing is, since AFAIU code is supposed to be ascii or maybe latin-1, with some idea that in the future it will be utf8. I just don't feel that I know what the problems might be, so have avoided unescaped unicode in non-comments.

Reply:
We should extend this to string literals, I think unicode characters in strings is as common as in comments.